### PR TITLE
Attempt to improve stability a bit: 

### PR DESCRIPTION
**Pull request by alibabajevich**
tracebacks due to network issues appears quite often, after each process freezes

### CODE REVIEW

**Comment by joncage** [comment]
It's a decent start but I'm thinking..

1. We ought to be raising proper exceptions rather than tuples (a bug from the original code I wrote).
2. It would be cleaner to just detect and handle the issue without exceptions at all instead of using them for control as your current patch does.
3. This whole section (not just your patch) could do with an overhaul. In particular, I think it would help to separate the EDSM lookups into a new pipeline so the display loads as quick as the game; Could have a spinny loading icon while the info is being gathered. Then as info comes in from EDSM, update the UI (or display an error indicating there's an EDSM connection issue).

**Comment by alibabajevich** [comment]
> It's a decent start but I'm thinking..
> 
> 1. We ought to be raising proper exceptions rather than tuples (a bug from the original code I wrote).
> 2. It would be cleaner to just detect and handle the issue without exceptions at all instead of using them for control as your current patch does.
> 3. This whole section (not just your patch) could do with an overhaul. In particular, I think it would help to separate the EDSM lookups into a new pipeline so the display loads as quick as the game; Could have a spinny loading icon while the info is being gathered. Then as info comes in from EDSM, update the UI (or display an error indicating there's an EDSM connection issue).

Hi. This is awesome tool. Thank you for it. 
I'm just patched it for myself to be able to use it on my trip to Colonia :) 

Bit disagree on 2, once you using _requests_  lib that throws bunch of [exceptions](https://requests.readthedocs.io/en/latest/api/#exceptions) on quite usual network issues, probability of it for 500 jumps very close to 100%, and since it unprocessed, any of them will crush Flask thread. Pity but that make tool nearly unusable.  
 
o7

**Review by joncage — COMMENTED** [review]
> disagree on 2, once you using requests lib that throws bunch of exceptions on quite usual network issues, probability of it for 500 jumps very close to 100%, and since it unprocessed, any of them will crush Flask thread. Pity but that make tool nearly unusable.

Sorry, bad wording on my part; I wasn't suggesting we shouldn't handle exceptions that _are_thrown, only that when we re-throw, that we throw a valid exception or handle that aspect without resorting to throwing then immediately catching an exception.

File: EDScoutCore/EDSMInterface.py [hunk 1]
```diff
@@ -34,11 +37,18 @@ def get_system(system_name):
...
-
-    if data.status_code != 200:
-        raise ("request returned bad response code %d" % (data.status_code))
+    retry_count = 10
+    data = None
+    while not data and retry_count:
+        try:
+            data = requests.get("https://www.edsm.net/api-v1/system", requestData, headers=headers)
+            if data.status_code != 200:
+                raise ("request returned bad response code %d" % (data.status_code))
```
[the remark this answers is not in the document]
Comment (joncage) [reply EDScoutCore/EDSMInterface.py:46]: We could potentially catch a more closely linked exception type too I think: https://requests.readthedocs.io/en/master/_modules/requests/exceptions/

**Comment by joncage** [comment]
Made some updates based on your proposal: https://github.com/joncage/ed-scout/pull/167 ...just need to add some handling in the web interface to highlight there are EDSM link issues.

File: EDScoutCore/EDSMInterface.py [hunk 2]
```diff
@@ -34,11 +37,18 @@ def get_system(system_name):
         "showCoordinates": 1,
         "showPrimaryStar": 1
     }
-
-    data = requests.get("https://www.edsm.net/api-v1/system", requestData, headers=headers)
-
-    if data.status_code != 200:
-        raise ("request returned bad response code %d" % (data.status_code))
+    retry_count = 10
```
[the remark this answers is not in the document]
Comment (alibabajevich) [reply EDScoutCore/EDSMInterface.py:40]: Yes. Usually that timeout exceptions, in most cases second request passes, rarely third. 10 is a bit overkill, just to be sure. Stalls i think are OK. Jumping in space it's anyway quite a long and meditative stuff :).
On the other hand 10 attemps with some little delay could hide from user cases when edsm is down (fast server restarts or something). 
However when it's down after 10 attempts, then it's reasonable to notify user to try later and kill the app

**Pull request closed without merging**